gui/tracks.lua

```diff
@@ -0,0 +1,180 @@
+    frame={b = 4, r = 4, w = 50, h = 12},
+    viewscreens= {
+        'dungeonmode/Default',
+    },
+}
+
+function TestOverlay:getLabel()
+    local pos = dfhack.world.getAdventurer().pos
+    local adv_x, adv_y, adv_z = pos.x, pos.y, pos.z
+    return ("Your position: %s, %s, %s"):format(
+        adv_x, adv_y, adv_z
+    )
+end
+
+function TestOverlay:init()
+    self.directional = false
+    self.show_yours = true
+    self.show_others = true
+    self:addviews{
+        widgets.Label{
+            frame = {l = 0, t = 0},
+            text = {{ text = self:callback('getLabel') }}
+        },
+        widgets.ToggleHotkeyLabel{
+            frame={l = 0, t = 2},
+            label='Show Directions',
+            key='CUSTOM_CTRL_D',
+            options={{value=true, label="Enabled"},
+                     {value=false, label="Disabled"}},
+            initial_option = self.directional,
+            on_change=function(option) self.directional = option end
+        },
+        widgets.ToggleHotkeyLabel{
+            frame={l = 0, t = 3},
+            label='Show Your Tracks',
+            key='CUSTOM_CTRL_Y',
+            options={{value=true, label="Enabled"},
+                     {value=false, label="Disabled"}},
+            initial_option = self.show_yours,
+            on_change=function(option) self.show_yours = option end
+        },
+        widgets.ToggleHotkeyLabel{
+            frame={l = 0, t = 4},
+            label='Show Other Tracks',
+            key='CUSTOM_CTRL_O',
+            options={{value=true, label="Enabled"},
+                     {value=false, label="Disabled"}},
+            initial_option = self.show_others,
+            on_change=function(option) self.show_others = option end
+        },
+    }
+end
+
+function TestOverlay:onRenderFrame(dc, rect)
+    TestOverlay.super.onRenderFrame(self, dc, rect)
+
+    for i, track in pairs(visible_tracks) do
+        if track.z ~= df.global.window_z then goto continue end
+        local track_xyz = {
+            x1 = track.x, x2 = track.x,
+            y1 = track.y, y2 = track.y,
+            z1 = track.z, z2 = track.z
+        }
+        local pen = others_track
+        if track.flags.yours then
+            if not self.show_yours then goto continue end
+            pen = yours_track
+        else
+            if not self.show_others then goto continue end
+        end
+        if track.flags.has_direction and self.directional then
+            pen = get_directional_pen(track.flags.direction)
+        end
+        guidm.renderMapOverlay(function() return pen end, track_xyz)
+        ::continue::
+    end
+end
+
+TestScreen = defclass(TestScreen, gui.ZScreen)
+TestScreen.ATTRS {
+    focus_path='testscreen',
+    pass_movement_keys=true,
+    pass_mouse_clicks=true,
+}
+
+function TestScreen:init()
+    local window = TestOverlay{}
+    self:addviews{
+        window,
+    }
+    onTimer()
+    repeatUtil.scheduleEvery(timerId, 1, 'ticks', onTimer)
```

Comment:
I suggest not using `repeatUtil` here. That makes cleanup messy. Instead, update in your own `render` method when the tick counter has incremented.